src/rov/rov_cameras/src/rov_cameras.cpp

```diff
@@ -25,11 +25,11 @@
 using namespace std::chrono_literals;
 
 // Testing Pipeline
-// const std::string pipeline = std::string("videotestsrc ! video/x-raw,format=RGBA ! appsink");
+const std::string pipeline = std::string("v4l2src device=/dev/video0 io-mode=2 ! video/x-raw,format=YUY2,width=320,height=240,framerate=30/1 ! videoconvert ! appsink");
```

Comment:
v4l2 is pretty chunky, but its also extremely well optimized (its in the good list of GStreamer plugins). May be worth investigating the ability to use nvv4l2camerasrc. Might put it in GPU memory instead of CPU. All I did was get rid of excessive nvvidconv and videoconverts. YUY2 is equivalent to YUYV (apparently obscure knowledge) and the camera is capable of outputting that directly (camera does all the hard work). OpenCV seems to not care about what format you give it and always returns a cv::Mat in BGR8 format. Unknown if there is a way to change this behaviour natively. Personally, don't think that the pipeline will get more efficient unless we ditch OpenCV and create our own GStreamer sink.